readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -106,10 +117,13 @@ def test_incremental_state_update_with_no_update(self):
         ]
 
         for build_env in build_envs:
+            #  __import__('pdb').set_trace()
             with build_env:
                 build_env.update_build(BUILD_STATE_CLONING)
+                # FIXME: This method raise an assert exception,
+                # but is captured by the context manager
```

Comment:
Also I found this hidden falling tests, the context manager covered it. We have two more thing to notice here:
- We are using  `mock.ANY` inside this dict, what I have understand this shouldn't work here.
- If we try to check the `mock_calls` outside the context manager the tests will fail anyway since we are passing a reference here, which is captured by the mock, i.e at the end we have two calls (one from the original update, and one from the `__exit__`) with the same reference to the dict (the values from the first call are lost).
https://github.com/rtfd/readthedocs.org/blob/e38d16ca1e05effac32bcf9de0175b7ba9085f20/readthedocs/doc_builder/environments.py#L573